internal/controllers/rollout/controller.go

```diff
@@ -65,6 +65,20 @@ func (c *controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 	}
 
+	// Cancel pending resynthesis for compositions that are ignoring side effects.
+	// Do this before calculating selecting the next composition to avoid
+	// skewing the results.
+	for _, comp := range comps.Items {
+		if comp.Status.PendingResynthesis != nil && comp.ShouldIgnoreSideEffects() {
+			comp.Status.PendingResynthesis = nil
+			err := c.client.Status().Update(ctx, &comp)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("clearing PendingResythesis field for %s in namespace %s: %w", comp.Name, comp.Namespace, err)
+			}
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
Curious why we need to requeue here?

Reply:
Good catch, we don't, we can rely on the write to kick off the next tick